src/bbmustache.erl

```diff
@@ -689,3 +707,73 @@ is_recursive_data(_)                                -> false.
 is_recursive_data([Tuple | _]) when is_tuple(Tuple) -> true;
 is_recursive_data(_)                                -> false.
 -endif.
+
+%% @doc Processes command-line commands (render, ...)
+-spec process_commands([getopt:option()], [string()]) -> ok.
+process_commands([], []) ->
+    help();
+process_commands([help | _], _) ->
+    help();
+process_commands([version | _], _Options) ->
+    {ok, AppConfig} = application:get_all_key(bbmustache),
+    Version = proplists:get_value(vsn, AppConfig),
+    io:format("~p~n", [Version]);
+process_commands([commands | _], _Options) ->
+    Commands = "render [file,...]",
+    io:format("~s~n", [Commands]);
+process_commands(Options, ["render" | TemplateFileNames]) ->
+    KeyType = proplists:get_value(key_type, Options, atom),
+    RenderOptions = [{key_type, KeyType}],
+    DataFileName = proplists:get_value(data_file, Options),
+    Data = merge_data_files(DataFileName),
+    lists:foreach(fun(TemplateFileName) ->
+                        {ok, Template} = file:read_file(TemplateFileName),
+                        Ret = render(Template, Data, RenderOptions),
+                        io:format("~s~n", [Ret])
+                  end, TemplateFileNames);
+process_commands(_, _) ->
+    throw(unrecognized_or_unimplemented_command).
+
+%% @doc Returns command-line options.
+-spec option_spec_list() -> [getopt:option_spec()].
+option_spec_list() ->
+[
+    %% {Name,   ShortOpt,   LongOpt,        ArgSpec,        HelpMsg}
+    {help,      $h,         "help",         undefined,      "Show this help information."},
+    {version,   $v,         "version",      undefined,      "Output the current bbmustache version"},
+    {commands,  undefined,  "commands",     undefined,      "Show available commands."},
+    {key_type,  $k,         "key-type",     atom,           "Key type (atom | binary | string)."},
+    {data_file, $d,         "data-file",    string,         "Erlang terms file."}
+].
+
+%% @doc Returns command-line help.
+-spec help() -> ok.
+help() ->
+    getopt:usage(option_spec_list(), escript:script_name()).
+
+% @doc Checks for terms inclusing in terms using format "file".
+-spec check_terms_inclusion(string(), [term()], [term()]) -> [term()].
+check_terms_inclusion(RelativeRoot, [FileName | T], Terms) when is_list(FileName) ->
+    IncludedTerms = merge_data_files(filename:join(RelativeRoot, FileName)),
+    check_terms_inclusion(RelativeRoot, T, IncludedTerms ++ Terms);
```

Comment:
I'm trying to emulate relx. I need the same behavior because the data files will be the same. We can put it behind a flag.

Reply:
I didn't know that feature, so I'll look it up.
However, bbmustache does not depend on relax, so it is not possible to support compatibility with relx.
I'm not always writing erlang at works, so it's impossible to follow it.

Reply:
Relx uses bbmustache to expand sys.config template from overlay vars stored in files as terms. It's my use case. I guess we could add a flag --merge-terms ?

Reply:
In my research, relx supports this syntax because Erlang's config (sys.config) supports it.
So I allow bbmustache to support this **without** any options.

> Relx also allows you to include other overlay files from an overlay file, the syntax is the same as used in Erlang (see config).

https://github.com/erlware/relx/wiki/Overlays

> There is, however, a syntax for sys.config that allows pointing out other .config files:
> [{Application, [{Par, Val}]} | File].

http://erlang.org/doc/man/config.html